web/docs/project/SecretGeneratorBlock.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from "react";
+import styles from "./SecretGeneratorBlock.module.css";
+
+interface Props {
+  buttonLabel?: string;
+}
+
+export function SecretGeneratorBlock({
+  buttonLabel = "Generate secret",
+}: Props) {
+  const [secret, setSecret] = useState<string | null>(null);
+
+  const generate = () => {
+    const bytes = crypto.getRandomValues(new Uint8Array(32));
+    const hex = Array.from(bytes, (b) => b.toString(16).padStart(2, "0")).join(
+      "",
+    );
+    setSecret(hex);
+  };
+
+  const copy = () => {
+    if (secret) navigator.clipboard?.writeText(secret).catch(() => {});
+  };
+
+  if (!secret) {
+    return (
+      <button onClick={generate} className={styles.generateBtn} type="button">
+        {buttonLabel}
+      </button>
+    );
+  }
+
+  return (
+    <span className={styles.container}>
+      <button
+        onClick={copy}
+        className={styles.iconBtn}
+        type="button"
+        title="Copy secret"
+      >
+        <svg
+          width="15"
+          height="15"
+          viewBox="0 0 15 15"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <path
+            d="M1 9.50006C1 10.3285 1.67157 11.0001 2.5 11.0001H4L4 10.0001H2.5C2.22386 10.0001 2 9.7762 2 9.50006L2 2.50006C2 2.22392 2.22386 2.00006 2.5 2.00006L9.5 2.00006C9.77614 2.00006 10 2.22392 10 2.50006V4.00002H5.5C4.67158 4.00002 4 4.67159 4 5.50002V12.5C4 13.3284 4.67158 14 5.5 14H12.5C13.3284 14 14 13.3284 14 12.5V5.50002C14 4.67159 13.3284 4.00002 12.5 4.00002H11V2.50006C11 1.67163 10.3284 1.00006 9.5 1.00006H2.5C1.67157 1.00006 1 1.67163 1 2.50006V9.50006ZM5 5.50002C5 5.22388 5.22386 5.00002 5.5 5.00002H12.5C12.7761 5.00002 13 5.22388 13 5.50002V12.5C13 12.7762 12.7761 13 12.5 13H5.5C5.22386 13 5 12.7762 5 12.5V5.50002Z"
+            fill="currentColor"
+            fillRule="evenodd"
+            clipRule="evenodd"
+          />
+        </svg>
```

Comment:
Extract this to a named component in this same file, so it's clearer what it is (e.g. `CopyIcon`). Also add a comment saying linking to where you took this icon from 🙏